llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1516,16 +1516,23 @@ static void transformRecipestoEVLRecipes(VPlan &Plan, VPValue &EVL) {
                         IntegerType::getInt1Ty(CI->getContext())));
                     Ops.push_back(Mask);
                     Ops.push_back(&EVL);
+                    FastMathFlags FMF = {};
+                    if (isa<FPMathOperator>(CI))
+                      FMF = CI->getFastMathFlags();
```

Comment:
FastMathFlags on cast instructions seems to be a new thing? https://github.com/llvm/llvm-project/pull/115894

In that case should we preserve them in VPWidenCastRecipe/VPScalarCastRecipe first? I can't find anything there that preserves them